source/ShopPanel.cpp

```diff
@@ -616,6 +619,50 @@ bool ShopPanel::Scroll(double dx, double dy)
 
 
 
+void ShopPanel::DoFind(const string &text)
+{
+	vector<Zone>::const_iterator it = zones.begin();
+	int bestIndex = 9999;
+	vector<Zone>::const_iterator bestItem;
+	int index;
+	for( ; it != zones.end(); ++it)
+	{
+		const Ship *ship = it->GetShip();
+		const Outfit *outfit = it->GetOutfit();
+		if(ship && HasItem(ship->VariantName()))
+			index = Search(ship->VariantName(), text);
+		else if(outfit && HasItem(outfit->TrueName()))
+			index = Search(outfit->TrueName(), text);
```

Comment:
Doesn't `HasItem` check only if something is for sale?

Also, variant / true names are internal, the player sees only display names.

Reply:
Ahh good catch for the display model names. I've changed the `Search()` functions to use `ship->DisplayModelName()` and `outfit->DisplayName()` respectively.

In regards to what `HasItem()` checks, for ships it only searches for ships on sale (since they're the only ones that appear in the shop view), and for outfits it searches for all that are on sale, in cargo, in storage, or installed. Should that be sufficient?